x-pack/plugins/security_solution/public/detections/hooks/trigger_actions_alert_table/use_cell_actions.tsx

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { EuiDataGridColumn, EuiDataGridRefProps } from '@elastic/eui';
+import type { TimelineNonEcsData } from '@kbn/timelines-plugin/common';
+import { get } from 'lodash';
+import type { EcsSecurityExtension as Ecs } from '@kbn/securitysolution-ecs';
+import type { AlertsTableConfigurationRegistry } from '@kbn/triggers-actions-ui-plugin/public/types';
+import { useCallback, useMemo } from 'react';
+import { tableDefaults } from '../../../common/store/data_table/defaults';
+import { VIEW_SELECTION } from '../../../../common/constants';
+import { useSourcererDataView } from '../../../common/containers/sourcerer';
+import { defaultCellActions } from '../../../common/lib/cell_actions/default_cell_actions';
+import type { TableId } from '../../../../common/types';
+import { FIELDS_WITHOUT_CELL_ACTIONS } from '../../../common/lib/cell_actions/constants';
+import { SourcererScopeName } from '../../../common/store/sourcerer/model';
+import { useShallowEqualSelector } from '../../../common/hooks/use_selector';
+import { dataTableSelectors } from '../../../common/store/data_table';
+
+export const getUseCellActionsHook = (
+  tableId: TableId
+): AlertsTableConfigurationRegistry['useCellActions'] => {
+  const useCellActions = ({
+    columns,
+    data,
+    ecsData,
+    dataGridRef,
+    pageSize,
+  }: {
+    // Hover Actions
+    columns: EuiDataGridColumn[];
+    data: unknown[][];
+    ecsData: unknown[];
+    dataGridRef?: EuiDataGridRefProps | null;
+    pageSize: number;
+  }) => {
+    const { browserFields } = useSourcererDataView(SourcererScopeName.detections);
+    /**
+     * There is difference between how `triggers actions` fetched data v/s
+     * how security solution fetches data via timelineSearchStrategy
+     *
+     * _id and _index fields are array in timelineSearchStrategy  but not in
+     * ruleStrategy
+     *
+     *
+     */
+    const finalData = (data as TimelineNonEcsData[][]).map((row) =>
+      row.map((field) => {
+        let localField = field;
+        if (['_id', '_index'].includes(field.field)) {
+          const newValue = field.value ?? '';
+          localField = {
+            field: field.field,
+            value: Array.isArray(newValue) ? newValue : [newValue],
+          };
+        }
+        return localField;
+      })
+    );
```

Comment:
Could we `useMemo` this? it is causing many updates and repeated calls to `getCellActions`

Reply:
IT is done now.